src/fee_lab_to_nwb/scherrer_ophys/scherrerophysimagingextractor.py

```diff
@@ -19,14 +19,19 @@ class ScherrerOphysImagingExtractor(ImagingExtractor):
     def __init__(self, file_path: FilePathType):
         super().__init__()
         self.file_path = file_path
-        self.video_capture_context = VideoCaptureContext(str(self.file_path))
+        self.video_capture_context = VideoCaptureContext(
+            str(self.file_path)
+        ).__enter__()
```

Comment:
It's weird to call `__enter__` directly without actually doing so in the proper `with` context - also, the `__init__` to the class should be designed to work without have to call `__enter__` to do any of that.

Was there a reason you needed to change this?

Reply:
The reason why I did that is to make sure the file is opened only once -- if `get_video` is called with chunking i.e. not from 0 to num_frames but only some range of frames, then using with context within the get_video and calling `next()` on the videocapture would mess up indexing. 

But again, this is might not be the best approach, I'm totally open for suggestions.

Reply:
Right, the rest of the approach here is fine treating it as an interable object outside of a `with` statement - so why do you need to call `__enter__` at all at this line since all it does is essentially what `__init__` does? (https://github.com/catalystneuro/neuroconv/blob/main/src/neuroconv/datainterfaces/behavior/movie/movie_utils.py#L134-L136)

Reply:
Checkout how the `MovieDataChunkIterator` handles this: https://github.com/catalystneuro/neuroconv/blob/main/src/neuroconv/datainterfaces/behavior/movie/movie_utils.py#L155 + https://github.com/catalystneuro/neuroconv/blob/main/src/neuroconv/datainterfaces/behavior/movie/movie_utils.py#L201

without needing to call `__enter__()` (which should only ever be called indirectly via `with ... as ...:`